lib/supavisor/client_handler.ex

```diff
@@ -154,14 +154,16 @@ defmodule Supavisor.ClientHandler do
         Logger.debug("ClientHandler: Client startup message: #{inspect(hello)}")
         {type, {user, tenant_or_alias, db_name}} = HH.parse_user_info(hello.payload)
 
-        not_allowed = ["\"", "\\"]
+        # Validate user and db_name according to PostgreSQL rules.
+        # The rules are: 1-63 characters, alphanumeric, underscore and $
+        rule = ~r/^[a-z_][a-z0-9_$]*$/
 
-        if String.contains?(user, not_allowed) or String.contains?(db_name, not_allowed) do
-          reason = "Invalid characters in user or db_name"
+        if !(user =~ rule && db_name =~ rule) do
```

Comment:
I preferred older form of:

```suggestion
        if not user =~ rule or not db_name =~ rule do
```

Though I would probably swap bodies of the `if` and use positive predicate there, so it would be:

```elixir
if user =~ rule and db_name =~ rule do
  # old `else` block content
else
  # old `if` block content
end
```

Reply:
100%, just need to swap cases 😄